src/zope/app/container/browser/tests/test_contents_functional.py

```diff
@@ -320,24 +318,42 @@ def test_unmoveable_object(self):
         body = response.text
         self.assertIn("cannot be moved", body)
 
-    @unittest.skipIf(str is not bytes, #Py2 only
-                     "Only Python 2 can do str(b'encoded') and get the right thing")
     def test_copy_then_delete_with_unicode_name(self):
         # Tests unicode on object copied then deleted (#238579)
-        # The zope.publisher.browser conversion methods for text/str
-        # rely on the default encoding, which breaks on this unicode name in Python 3.
-        # We either wind up with "b'voil\\xe0'" or a list of the ints that make
-        # up the bytes, depending on which order we try for the type names.
+
+        # The 'Accept-Charset' is important on Python 3 to get
+        # the encoding correct. The query string is actually correctly
+        # decoded to unicode (if it's ever even *encoded*) on the
+        # receiving side, but
+        # zope.publisher.browser.BrowserRequest._decode assumes it was
+        # smuggled as latin-1, and thus encodes it back to those
+        # bytes. If we don't specify latin-1 as a charset
+        # zope.publsher.http.HTTPCharsets will only let it try to be
+        # decoded as utf-8, which doesn't work.
+        # On Python 2, if we don't encode the parameter, it's improperly
+        # decoded before it ever gets to _decode, arriving as 'voill?'.
+
+        # This is most likely a mismatch
+        # somewhere in the webtest.TestApp/zope.app.wsgi.testlayer/zope.publisher
+        # stack.
 
         # create a file with an accentuated unicode name
         root = self.getRootFolder()
         root[u'voil\xe0'] = File()
         transaction.commit()
 
         # copy the object
-        response = self.publish('/@@contents.html', basic='mgr:mgrpw', form={
-            'ids:list' : (u'voil\xe0'.encode('utf-8'),),
-            'container_copy_button' : '' })
+        response = self.publish(
+            '/@@contents.html',
+            basic='mgr:mgrpw',
+            form={
+                'ids:list' : u'voil\xe0'.encode('utf-8'),
+                'container_copy_button' : ''
```

Comment:
No space before `:` please.  (Yes, I see it was present in the old code.)

Also I like a trailing comma for multiline dict literals when the closing `}` gets a line of its own.